src/values/ArrayValue.js

```diff
@@ -26,9 +26,15 @@ import { type OperationDescriptor } from "../utils/generator.js";
 import invariant from "../invariant.js";
 import { NestedOptimizedFunctionSideEffect } from "../errors.js";
 import { PropertyDescriptor } from "../descriptors.js";
+import { SimpleNormalCompletion } from "../completions.js";
 
+type ArrayNestedOptimizedFunctionType = "map" | "filter";
```

Comment:
Is `filter` used anywhere? I wasn't sure why we have this now when it doesn't really appear to be used anywhere?

Reply:
I put it in as a reminder for whoever adds support for `filter` (in case it's not me) that the leaking semantics of these various operators are different. So for instance, a non-mutating filter cannot alias. I was also planning on adding it soon.